### PR TITLE
AutoPlugin everywhere

### PR DESCRIPTION
**Pull request by guersam**
This PR follows #244 introducing more API changes. README is not updated yet and there is a room for further discussion. The main idea is to make each existing container (jetty, tomcat) an AutoPlugin with a dedicated config like this:

```scala
// build.sbt
enablePlugins(JettyPlugin)

// in SBT console
> jetty:start
```

Custom container settings can be done by using `ContainerPlugin` with config `Container`:

```scala
// build.sbt
enablePlugins(ContainerPlugin)

containerLibs in Container := Seq(
    "org.eclipse.jetty" %  "jetty-webapp" % "9.1.0.v20131115"
  , "org.eclipse.jetty" %  "jetty-plus"   % "9.1.0.v20131115"
  , "test"              %% "runner"       % "0.1.0-SNAPSHOT"
  )

containerLaunchCmd in Container :=
  Seq("runner.Run", "8080", (target in webappPrepare).value.absolutePath)

// in SBT console
> container:start
```

### CODE REVIEW

**Comment by earldouglas** [comment]
Awesome, thanks!  Give me a bit to review, but it looks great at a glance!

File: src/main/scala/com/earldouglas/xwp/ContainerPlugin.scala [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    val containerLibs        = settingKey[Seq[ModuleID]]("container libraries")
+    val containerMain        = settingKey[String]("container main class")
+    val containerPort        = settingKey[Int]("port number to be used by container")
+    val containerConfigFile  = settingKey[Option[File]]("path of container configuration file")
+    val containerArgs        = settingKey[Seq[String]]("additional container args")
+    val containerLaunchCmd   = settingKey[Seq[String]]("command to launch container")
+    val containerForkOptions = settingKey[ForkOptions]("fork options")
+  }
+
+  private lazy val containerInstance = settingKey[AtomicReference[Option[Process]]]("Current container process")
```
Comment (guersam) [inline src/main/scala/com/earldouglas/xwp/ContainerPlugin.scala]: I'm not 100% confident of this, but tests pass either on my local machine and travis.

**Comment by earldouglas** [comment]
Pardon the delay -- super busy these weeks!

**Comment by guersam** [comment]
no problem, it's not urgent to me :)

**Pull request merged**